src/main/java/org/igniterealtime/openfire/plugins/pushnotification/Push0IQHandler.java

```diff
@@ -210,4 +221,91 @@ public IQHandlerInfo getInfo()
     {
         return new IQHandlerInfo( ELEMENT_NAME, ELEMENT_NAMESPACE );
     }
+
+    @Override
+    public Iterator<Element> getIdentities(String name, String node, JID senderJID) {
+        Element identity = DocumentHelper.createElement("identity");
+        identity.addAttribute("category", "pubsub");
+        identity.addAttribute("type", "push");
+        return Collections.singleton(identity).iterator();
+    }
+
+    @Override
+    public Iterator<String> getFeatures()
+    {
+        return Collections.singleton( ELEMENT_NAMESPACE ).iterator();
+    }
+
+    @Override
+    public Iterator<String> getFeatures(String name, String node, JID senderJID) {
+        return Arrays.asList(ELEMENT_NAMESPACE, "jabber:x:data").iterator();
+    }
+    @Override
+    public DataForm getExtendedInfo(String name, String node, JID senderJID) {
+        return null;
+    }
+
+    @Override
+    public boolean hasInfo(String name, String node, JID senderJID) {
+        return true;
+    }
+    @Override
+    public Iterator<DiscoItem> getItems(String name, String node, JID senderJID) {
+        List<DiscoItem> answer = new ArrayList<>();
+
+        if (!ELEMENT_NAMESPACE.equals(node)) {
+            answer = Collections.emptyList();
+        }
+        else {
+            final User user;
+            try
+            {
+                user = XMPPServer.getInstance().getUserManager().getUser( senderJID.getNode() );
```

Comment:
If I read this correctly, doing a service discovery items request would return personalized (as opposed to generic) information. I'm not sure if that's how disco/items is to be used. Where did you find this usage pattern?